### ISSUE TITLE
Warning: Undefined array key "post_type" in /plugins/woocommerce/includes/tracks/events/class-wc-products-tracking.php on line 316

### ISSUE THREAD

**Issue by jacoswan** [woothemes/woocommerce#34652]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When WooCommerce is active and a new Post is created, the following shows briefly before the Post editor is displayed: 

```
Warning: Undefined array key "post_type" in /Users/woohoo/Sites/woo/wp-content/plugins/woocommerce/includes/tracks/events/class-wc-products-tracking.php on line 316
```

This happens with at least WC 6.8.2 and WC 6.9.0-rc.1

### Expected behavior

Would expect no warning message as it the case when creating Pages, Products, or other types. is

### Actual behavior

A warning is observed. 

### Steps to reproduce

1. Enabled only WooCommerce (and Storefront theme)
2. Go WP Admin > Posts > Add New
3. Observe the message displaying briefly prior to the editor loading

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://woo.tst
Site address (URL): https://woo.tst
WC Version: 6.9.0
REST API Version: ✔ 6.9.0
WC Blocks Version: ✔ 8.3.2
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.54 (Unix) OpenSSL/1.1.1q PHP/8.1.8
PHP Version: 8.1.8
PHP Post Max Size: 500 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.84.0
(SecureTransport) OpenSSL/1.1.1q

SUHOSIN Installed: –
MySQL Version: 10.8.3-MariaDB
Max Upload Size: 500 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.9.0
WC Database Prefix: wp_
Total Database Size: 11.20MB
Database Data Size: 7.70MB
Database Index Size: 3.50MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.11MB + Index: 0.13MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.23MB + Index: 0.17MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.16MB + Index: 0.16MB + Engine InnoDB
wp_automatewoo_abandoned_carts: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_customers: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_automatewoo_customer_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_events: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_guests: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_automatewoo_guest_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_log_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_queue: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_queue_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.08MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 5.05MB + Index: 0.14MB + Engine InnoDB
wp_postmeta: Data: 0.45MB + Index: 0.45MB + Engine InnoDB
wp_posts: Data: 0.17MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availability: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availabilitymeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_booking_relationships: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_bundle_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_order_composite_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points_log: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_bis_activity: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_bis_notifications: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_woocommerce_bis_notificationsmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_bundled_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_bundled_items: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_exported_csv_items: Data: 0.08MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_gc_activity: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_gc_cards: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_gc_cardsmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_table_rates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

admin_login_history: 3
attachment: 34
aw_workflow: 3
event_ticket: 9
global_product_addon: 2
nav_menu_item: 8
page: 16
post: 9
product: 71
product_variation: 11
revision: 19
shipping_package: 3
shop_coupon: 3
shop_order: 57
shop_order_refund: 6
shop_subscription: 1
wbcr-snippets: 10
wc_booking: 2
wp_global_styles: 2
wp_navigation: 1
wp_template: 1
wp_template_part: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (2) ###

WooCommerce Beta Tester: by WooCommerce – 2.0.5
WooCommerce: by Automattic – 6.9.0-rc.1

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: accommodation-booking (accommodation-booking)
booking (booking)
bundle (bundle)
composite (composite)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude_from_catalog (exclude_from_catalog)
exclude_from_search (exclude_from_search)
exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.1.3
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
experimental-product-tour
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2022-09-14 06:14:14 +00:00
Options: ✔
Notes: 42
Onboarding: completed

### Action Scheduler ###

Canceled: 1
Oldest: 2022-09-13 06:44:02 +0000
Newest: 2022-09-13 06:44:02 +0000

Complete: 566
Oldest: 2022-08-15 04:58:28 +0000
Newest: 2022-09-13 06:44:02 +0000

Pending: 4
Oldest: 2022-09-14 04:56:35 +0000
Newest: 2022-10-02 11:22:34 +0000

### Status report information ###

Generated at: 2022-09-13 06:48:17 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by swatipawarGS** [comment]
Hi @jacoswan,

Thank you for taking the time to report this bug, we really appreciate your help. We are unable to reproduce the issue on our end using the WooCommerce 6.9.0 version, PHP 8.0.22 version, and WordPress 6.0.2 version. 

Please find below the screencast for reference :

https://user-images.githubusercontent.com/94531721/190077918-fb464fd3-5799-40eb-a732-7a55f629eba1.mp4

Please provide us with more details and any additional steps required to repro this issue which may help us evaluate it further.

**Comment by daniyalahmadk** [comment]
I was able to replicate the issue as well on WC 6.9.1 with PHP 8.0:

https://d.pr/i/MzASbJ

It's missing the following check on that line of code:

`isset( $_GET['post_type'] )`

**Comment by kaushikasomaiya** [comment]
For me, only occurs if

- Debug mode is on
- Usage tracking is allowed

Workaround:  Untick `Enable tracking` under **WooCommerce > Settings > Advanced > WooCommerce.com**

**Comment by swatipawarGS** [comment]
Hi @jacoswan, @daniyalahmadk,

Thank you for the updated video. We are able to reproduce the issue on our end with 6.9.2 version, PHP 8.0.22 version, and WordPress 6.0.2 version. 

Please find below the screencast for reference :

https://user-images.githubusercontent.com/94531721/190581202-52d3b577-ff61-4809-9196-d0d1ff1bab43.mp4

Priority of this issue will be assigned shortly after we’d go through a round of prioritization. Please note it may take a couple of days.

**Comment by tammullen** [comment]
I was able to replicate this in 7.0 RC, only when tracking is enabled.
`Warning:  Undefined array key "post_type" in /plugins/woocommerce/includes/tracks/events/class-wc-products-tracking.php on line 320`

**Comment by kaushikasomaiya** [comment]
5696368-zen

**Comment by JiveDig** [comment]
We just hit this today, and turning off tracking resolved it for now as well.

**Comment by kaushikasomaiya** [comment]
5815312-zen

**Comment by peterfabian** [comment]
I think this was fixed with https://github.com/woocommerce/woocommerce/pull/34501 and released in WC 7.3.0

Please feel free to comment/reopen if not. Thanks!
